File: Irrgarten/src/irrgarten/TestP2.java

package irrgarten;

/**
 *
 * @author leonin04
 */

public class TestP2 {
    static void pruebaMonster(){
        Monster m = new Monster("Monstruito", 10, 10);
        System.out.println(m.toString());
        System.out.println("¿Muerto? " + m.dead());
        System.out.println("Daño: " + m.attack());
        System.out.println("Colocando monstruo en el (1,1)...");
        m.setPos(1, 1);
        System.out.println(m.toString());
        System.out.println("Matando monstruo...");
        m.gotWounded();
        System.out.println(m.toString());
        m.gotWounded();
        m.gotWounded();
        m.gotWounded();
        m.gotWounded();
        System.out.println(m.toString());
        System.out.println("¿Muerto? " + m.dead());
    }
    
    static void pruebaPlayer(){
        Player p = new Player('1',10,10);
        System.out.println(p.toString());
        System.out.println("Colocando en 0,0 ");
        p.setPos(0,0);
        System.out.println("Colocado en: (" + p.getRow() + "," + p.getCol() + ")");
        System.out.println("¿Muerto? " + p.dead());
        System.out.println("Atacando: " + p.attack());
        //System.out.println("Defendiendo: " + p.defend(5));        
    }
    
    static void pruebaLabyrinth(){
        Labyrinth l = new Labyrinth(10,10,1,1);
        System.out.println("¿Hay ganador? " + l.haveAWinner());
        System.out.println("Pintando laberinto: ");
        System.out.println(l.toString());
        Monster m = new Monster ("Ares",10,10);
        System.out.println("Poniendo monstruo en (2,2)");
        l.addMonster(2, 2, m);
        System.out.println(l.toString());

    }
    
    public static void main(String[] args) {
       System.out.println("Probando monstruo: ");
       pruebaMonster();
       
       System.out.println("Probando player: ");
       pruebaPlayer();
       
       System.out.println("Probando labyrinth: ");
       pruebaLabyrinth();

    }
    
}

File: Irrgarten_Ruby/Irrgarten_Ruby_P4/test_p2.rb
#encoding:utf-8

require_relative 'Monster'
require_relative 'Player'
require_relative 'Labyrinth'
require_relative 'Game'
require_relative 'GameState'

module Irrgarten
	class TestP2
		def self.prueba_monster()
			m = Monster.new("Monstruito",10,10)
			puts m.to_s
			puts "¿Muerto? #{m.dead}"
			puts "Daño: #{m.attack}"
			puts "Colocando monstruo en el (1,1)..."
			m.set_pos(1,1)
			puts m.to_s
			puts "Matando monstruo..."
			m.got_wounded
			puts m.to_s
			m.got_wounded
			m.got_wounded
			m.got_wounded
			m.got_wounded
			puts m.to_s
			puts "¿Muerto? #{m.dead}"
		end
		
		def self.prueba_player()
			p = Player.new('1', 10,10)
			puts p.to_s
			puts "Colocando en 0,0:"
			p.set_pos(0,0)
			puts "Colocado en: (#{p.row},#{p.col})"
			puts "¿Muerto? #{p.dead}"
			puts "Atacando: #{p.attack}"
			#puts "Defendiendo #{p.defend}"
		end
		
		def self.prueba_labyrinth()
			l = Labyrinth.new(10,10,1,1)
			puts "¿Hay ganador? #{l.have_a_winner}"
			puts "Pintando laberinto:"
			puts l.to_s
			m = Monster.new("Ares",10,10)
			puts "Poniendo monstruo en (2,2)"
			l.add_monster(2,2,m)
			puts l.to_s
		end
		
		def self.prueba_game()
			g = Game.new(3)
			puts "¿Hay ganador? #{g.finished}"
			gs = g.get_game_state
			puts "#{gs.labyrinth}"
		end
	end
end

puts "Probando Monster"
Irrgarten::TestP2.prueba_monster
puts "\n Probando Player"
Irrgarten::TestP2.prueba_player
puts "\n Probando Labyrinth"
Irrgarten::TestP2.prueba_labyrinth
puts "\n Probando Game"
Irrgarten::TestP2.prueba_game
